sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/SetLocaleForDataTypes.cs

```diff
@@ -127,17 +127,14 @@ private static SyntaxNode GetAssignmentTargetVariable(ObjectCreationExpressionSy
         {
             var parent = objectCreation.GetSelfOrTopParenthesizedExpression().Parent;
 
-            var assignment = parent as AssignmentExpressionSyntax;
-            if (assignment != null)
+            if (parent is AssignmentExpressionSyntax assignment)
             {
                 return assignment.Left;
             }
 
-            var equalsClause = parent as EqualsValueClauseSyntax;
-            if (equalsClause != null)
+            if (parent is EqualsValueClauseSyntax equalsClause)
             {
-                var variableDeclaration = equalsClause.Parent.Parent as VariableDeclarationSyntax;
-                if (variableDeclaration != null)
+                if (equalsClause.Parent.Parent is VariableDeclarationSyntax variableDeclaration)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1066)